HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/VMGalleryVMCreationOperation.cs

```diff
@@ -158,16 +152,29 @@ public void Report(IOperationReport value)
 
     private void UpdateProgress(IOperationReport report, string localizedKey, string fileName)
     {
-        var bytesReceivedSoFar = BytesHelper.ConvertBytesToString(report.BytesReceived);
-        var totalBytesToReceive = BytesHelper.ConvertBytesToString(report.TotalBytesToReceive);
-        var progressPercentage = (uint)((report.BytesReceived / (double)report.TotalBytesToReceive) * 100D);
-        var displayString = _stringResource.GetLocalized(localizedKey, fileName, $"{bytesReceivedSoFar}/{totalBytesToReceive}");
-        Progress?.Invoke(this, new CreateComputeSystemProgressEventArgs(displayString, progressPercentage));
+        var bytesReceivedSoFar = BytesHelper.ConvertBytesToString((ulong)report.ProgressObject.BytesReceived);
```

Comment:
`BytesReceived` is stored as a `long` but here it is casted into a `ulong`?  Why not store the variable as `ulong`?

Reply:
Since this is only ever casted as a ulong for this particular method, I think I'll leave it as is for now.